### ISSUE TITLE
App becomes mostly unresponsive on Windows when left unattended for hours

### ISSUE THREAD

**Issue by senkwe** [pawelsalawa/sqlitestudio#5132]
### Details
App becomes mostly unresponsive on Windows when left unattended for hours (eg overnight)

This happens with regularity across two different machines running Windows 11. If I leave the app open overnight, I'll come back in the morning to find it completely unresponsive. I have to kill the process and open a new instance.

### Steps to reproduce
Leave open overnight

### Operating system
Windows 11

### SQLiteStudio version
3.4.4

https://github.com/user-attachments/assets/a8f42161-4b2e-453f-85ba-0e1c5ac1d294

**Comment by pawelsalawa** [comment]
Is any of connected databases located on a network shared folder/drive?

**Comment by pawelsalawa** [comment]
Okay, I will try to reproduce it today.

**Comment by pawelsalawa** [comment]
This night it did not happen. After ~8 hours the app works as expected.

I guess there is some other factor causing this. Is there another application modifying database that is open in your SQLiteStudio over the night? Of course it should not break SQLiteStudio like that, but I'm looking for anything that could give me some hints.

**Comment by senkwe** [comment]
Nope, nothing at all. It's a minor inconvenience so not too worried about it. 

And I seem to be the only seeing this issue. But it's strange that it's not specific to a particular machine.

**Comment by pawelsalawa** [comment]
I understand. Still I would be happy to nail it down - if possible. If we fail to find the cause, then I will surrender ;) Still, let's give it a try.

So...
1. Is this 100% reproducible for you? I mean does this happen every single time when you leave it for several hours?
2. Have you ever seen for example that the application freezed after 1 hour being unattended?
3. Do you have SQL Editor windows open when such freeze happens? Approximately - how many?
4. Do you use any SQLite extensions dynamically loaded to databases? (either through `load_extension()` function or through SQLiteStudio's Extension Manager window)

**Comment by pawelsalawa** [comment]
And one more thing. Could you please next time start SQLiteStudio from command line, adding debug options, so the next time we may have some useful information when it freezes? I mean:
```
SQLiteStudio.exe -d --debug-file=C:\path\to\logfile.txt
```
...just make sure that directory `C:\path\to\` that you choose exists and is writable.

**Comment by senkwe** [comment]
> 1. Is this 100% reproducible for you? I mean does this happen every single time when you leave it for several hours?

- Pretty much YES. Every once in a while it surprisingly doesn't happen.

> 2. Have you ever seen for example that the application freezed after 1 hour being unattended?

- NO. I do wonder if it has anything to do with the laptop going into hibernation. But if I leave it for an hour or so it's fine. This issue only seems to happen when I'm away from my computer for an extended period (when I go to bed at night)

> 3. Do you have SQL Editor windows open when such freeze happens? Approximately - how many?

- YES, I usually only have one editor open and perhaps 3 table views lately as my databases are tiny (literally no more than 9 rows in the largest table)

> 4. Do you use any SQLite extensions dynamically loaded to databases? (either through `load_extension()` function or through SQLiteStudio's Extension Manager window)

- NO

> 5. What RAM and CPU utilization does the Windows Task Manager show for the `SQLiteStudio.exe` process when it's freezed like that?

- Will report back on this

**Comment by senkwe** [comment]
> And one more thing. Could you please next time start SQLiteStudio from command line, adding debug options, so the next time we may have some useful information when it freezes? I mean:
> 
> ```
> SQLiteStudio.exe -d --debug-file=C:\path\to\logfile.txt
> ```
> 
> ...just make sure that directory `C:\path\to\` that you choose exists and is writable.

Great, will do.

**Comment by pawelsalawa** [comment]
Hibernation... I will try that.

**Comment by pawelsalawa** [comment]
I've tried with the hibernation, screen locking. Nothing causes this issue for me :( I'm suspending this for now. If any new hints appear, I will get back to it.

Thanks for reporting this anyway. It's important to know that there is such an issue. I just need to find reliable way to reproduce it.

**Comment by jjoosse** [comment]
I tried the development version and could not reproduce it. After switching off and on the external monitor or disconnecting and reconnecting the window was restored correctly.

**Comment by pawelsalawa** [comment]
Okay then. I mark it as resolved in 4.0.0.
